### PR TITLE
fix: update kusion doc to correct developer role

### PR DESCRIPTION
**Pull request by SS-Yuki**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):
docs/kusion/3-concepts/3-module/3-app-dev-guide.md

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by SS-Yuki** [comment]
I have read the CLA Document and I hereby sign the CLA

File: docs/kusion/3-concepts/3-module/3-app-dev-guide.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ To follow this guide, you will need:
 
 ## Workflow
 
-As a platform engineer, the workflow of developing a Kusion module looks like this:
+As an application developer, the workflow of developing a Kusion module looks like this:
```
Comment (adohe) [inline docs/kusion/3-concepts/3-module/3-app-dev-guide.md:12]: developing -> using ?

**Comment by adohe** [comment]
@SS-Yuki Thanks for contribution, I just leave some comments, ptal

File: docs/kusion/3-concepts/3-module/3-app-dev-guide.md (hunk 1 above)
Comment (SS-Yuki) [reply docs/kusion/3-concepts/3-module/3-app-dev-guide.md]: My apologies for the oversight. It has been updated.

**Comment by SS-Yuki** [comment]
> @SS-Yuki Thanks for contribution, I just leave some comments, ptal

Thank you very much for your reply. I am learning how to contribute on GitHub.

**Review by adohe — APPROVED** [review]
/lgtm

**Comment by adohe** [comment]
@SS-Yuki Thanks again ! We sincerely welcome each new contributor, and hope you enjoy~

**Pull request merged**